packages/insomnia-app/app/ui/components/dropdowns/remote-workspaces-dropdown.tsx

```diff
@@ -11,65 +8,64 @@ import * as models from '../../../models';
 import { database as db } from '../../../common/database';
 import { showAlert } from '../modals';
 import { strings } from '../../../common/strings';
+import { useSelector } from 'react-redux';
+import { selectActiveSpace, selectAllWorkspaces } from '../../redux/selectors';
+import { isWorkspace } from '../../../models/helpers/is-model';
+import { isLoggedIn } from '../../../account/session';
+import { isNotNullOrUndefined } from '../../../common/misc';
 
 interface Props {
   className?: string;
   vcs?: VCS | null;
-  workspaces: Workspace[];
 }
 
-interface State {
-  loading: boolean;
-  localProjects: Project[];
-  pullingProjects: Record<string, boolean>;
-  remoteProjects: Project[];
-}
-
-@autoBindMethodsForReact(AUTOBIND_CFG)
-class RemoteWorkspacesDropdown extends Component<Props, State> {
-  state: State = {
-    loading: false,
-    localProjects: [],
-    pullingProjects: {},
-    remoteProjects: [],
-  }
-
-  async _refreshRemoteWorkspaces() {
-    const { vcs } = this.props;
-
-    if (!vcs) {
+const useRemoteWorkspaces = (vcs?: VCS) => {
+  // Fetch from redux
+  const workspaces = useSelector(selectAllWorkspaces);
+  const activeSpace = useSelector(selectActiveSpace);
+  const spaceRemoteId = activeSpace?.remoteId || undefined;
+  const isRemoteSpace = isNotNullOrUndefined(spaceRemoteId);
+  const spaceId = activeSpace?._id;
+
+  // Local state
+  const [loading, setLoading] = useState(false);
+  const [localProjects, setLocalProjects] = useState<Project[]>([]);
+  const [remoteProjects, setRemoteProjects] = useState<Project[]>([]);
+  const [pullingProjects, setPullingProjects] = useState<Record<string, boolean>>({});
+
+  // Refresh remote spaces
+  const refresh = useCallback(async () => {
+    if (!vcs || !isLoggedIn()) {
       return;
     }
 
-    if (!session.isLoggedIn()) {
-      return;
-    }
-
-    this.setState({
-      loading: true,
-    });
-    const remoteProjects = await vcs.remoteProjects();
-    const localProjects = await vcs.localProjects();
-    this.setState({
-      remoteProjects,
-      localProjects,
-      loading: false,
-    });
-  }
-
-  async _handlePullRemoteWorkspace(project: Project) {
-    const { vcs } = this.props;
-
+    setLoading(true);
+    setRemoteProjects(await vcs.remoteProjects(spaceRemoteId));
+    setLocalProjects(await vcs.localProjects());
+    setLoading(false);
```

Comment:
Since we're now setting state in each operation here, rather then collecting it and then setting the state once once all operations succeeded, there *might* be some edge cases where we get partial/invalid state if an operation throws an error.

Not sure if we should opt for something like this as a bit of a defensive measure?
```suggestion
    setLoading(true);
    const remoteProjects = await vcs.remoteProjects(spaceRemoteId); 
    const localProjects = await vcs.localProjects();
    setRemoteProjects(remoteProjects);
    setLocalProjects(localProjects);
    setLoading(false);
```

Reply:
Good catch, this seems like a reasonable thing approach to take